song-server/src/main/java/bio/overture/song/server/service/analysis/AnalysisServiceSender.java

```diff
@@ -139,8 +144,9 @@ public Analysis securedDeepRead(@NonNull String studyId, String id) {
   }
 
   private void sendAnalysisMessage(
-      String studyId, String analysisId, AnalysisStates analysisState) {
-    val message = createAnalysisMessage(analysisId, studyId, analysisState, songServerId);
+      String studyId, String analysisId, AnalysisStates analysisState, AnalysisActions action) {
+    val analysis = internalAnalysisService.unsecuredDeepRead(analysisId);
```

Comment:
this will definetly decrease performance. For every publish/create/supress/unpublish action, there will be double the reading (and therefore double the transactions) since under the hood, `unsecuredDeepRead` is called for each action. 

I think the following partial refactor would eliminate the use of this extra call:
1. modify `AnalysisServiceImpl.checkedUpdateState` to return the analysis object
2. modify the `AnalysisService` interface to return an Analysis object for the publish, unpublish, suppress, and create methods
3. push the "presentation" related stuff (i.e ResponseEntity<String>) that the AnalysisServiceImpl.publish previous returned, to the controller. Likewise with create, unpublish and suppress

This is a bit of work, but will avoid any impact on performance.